### PR TITLE
feat: add Response assertions

### PR DESCRIPTION
**Pull request by mxschmitt**
Fixes https://github.com/microsoft/playwright-python/issues/1104
Backports https://github.com/microsoft/playwright/pull/10954/files

### CODE REVIEW

File: tests/async/test_browsercontext_add_cookies.py [hunk 1]
```diff
@@ -372,21 +370,5 @@ async def test_should_not_block_third_party_cookies(
...
-                "name": "username",
-                "path": "/",
-                "sameSite": "Lax" if is_chromium else "None",
-                "secure": False,
-                "value": "John Doe",
-            }
-        ]
-    else:
-        assert cookies == []
+    assert cookies == []
```
Comment (dgozman) [inline tests/async/test_browsercontext_add_cookies.py:374]: Why this change? Was there a similar one upstream?

File: tests/async/test_browsercontext_cookies.py [hunk 2]
```diff
@@ -39,12 +39,14 @@ async def test_should_get_a_cookie(context, page, server, is_chromium):
             "expires": -1,
             "httpOnly": False,
             "secure": False,
-            "sameSite": "Lax" if is_chromium else "None",
+            "sameSite": "Lax" if (is_chromium or is_firefox) else "None",
```
Comment (dgozman) [inline tests/async/test_browsercontext_cookies.py:42]: Same question here.

**Review by dgozman — APPROVED** [review]

File: tests/async/test_browsercontext_add_cookies.py (hunk 1 above)
Comment (mxschmitt) [reply tests/async/test_browsercontext_add_cookies.py:374]: Yes see the linked upstream backport.

File: tests/async/test_browsercontext_cookies.py (hunk 2 above)
Comment (mxschmitt) [reply tests/async/test_browsercontext_cookies.py:42]: Ditto
